compiler/rustc_typeck/src/astconv/generics.rs

```diff
@@ -39,7 +39,7 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
         );
 
         if let GenericParamDefKind::Const { .. } = param.kind {
-            if let GenericArg::Type(hir::Ty { kind: hir::TyKind::Infer, .. }) = arg {
+            if matches!(arg, GenericArg::Type(hir::Ty { kind: hir::TyKind::Infer, .. })) {
```

Comment:
we might want to add an optional help message recommending `generic_arg_infer` if we're on nightly